internal/spec/spec.go

```diff
@@ -140,6 +140,14 @@ type extraSpecs struct {
 	SourceSnapshot  string                      `json:"source_snapshot,omitempty" jsonschema:"description=The source snapshot to create this disk."`
 	SSHKeys         []string                    `json:"ssh_keys,omitempty" jsonschema:"description=A list of SSH keys to be added to the instance. The format is USERNAME:SSH_KEY"`
 	EnableBootDebug *bool                       `json:"enable_boot_debug,omitempty" jsonschema:"description=Enable boot debug on the VM."`
+	// Explicitly define fields from CloudConfigSpec for schema generation and direct access
+	DisableUpdates        *bool                  `json:"disable_updates,omitempty" jsonschema:"description=Disable OS updates on boot."`
+	ExtraContext          map[string]interface{} `json:"extra_context,omitempty" jsonschema:"description=Extra context for cloud-init templates."`
+	PreInstallScripts     map[string]string      `json:"pre_install_scripts,omitempty" jsonschema:"description=Scripts to run before runner installation."`
+	PostInstallScripts    map[string]string      `json:"post_install_scripts,omitempty" jsonschema:"description=Scripts to run after runner installation."`
+	RunnerInstallTemplate *string                `json:"runner_install_template,omitempty" jsonschema:"description=Custom runner installation script template."`
+	Files                 []cloudconfig.File     `json:"files,omitempty" jsonschema:"description=Files to be written to the instance."`
```

Comment:
I don't think this is used anywhere. While the `File{}` type can be instantiated directly and added to the cloud config before we render it, I think it's easier to just use the [`AddFile()`](https://github.com/cloudbase/garm-provider-common/blob/8ef03502d06e8e4790def8d7f1cea8feb913e4ca/cloudconfig/cloudconfig.go#L149) function attached to the [`CloudInit{}`](https://github.com/cloudbase/garm-provider-common/blob/8ef03502d06e8e4790def8d7f1cea8feb913e4ca/cloudconfig/cloudconfig.go#L70-L81) type.

I think we can remove this as well.